From ae444bf877fd94256a110d03582ddb045c541525 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Fri, 25 Feb 2022 09:25:31 -0800
Subject: [PATCH 1/2] nl80211: fix attribute definition for older kernels

Add compile time flags around the definition of
NL80211_ATTR_EHT_CAPABILITY to support kernels older than 5.10.0.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 net/wireless/nl80211.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -802,8 +802,14 @@ static const struct nla_policy nl80211_p
 	[NL80211_ATTR_MBSSID_ELEMS] = { .type = NLA_NESTED },
 	[NL80211_ATTR_RADAR_BACKGROUND] = { .type = NLA_FLAG },
 	[NL80211_ATTR_AP_SETTINGS_FLAGS] = { .type = NLA_U32 },
+#if LINUX_VERSION_IS_GEQ(5,10,0)
 	[NL80211_ATTR_EHT_CAPABILITY] =
 		NLA_POLICY_BINARY_RANGE(NL80211_EHT_MIN_CAPABILITY_LEN, NL80211_EHT_MAX_CAPABILITY_LEN),
+#else
+	[NL80211_ATTR_EHT_CAPABILITY] =
+		{ .type = NLA_BINARY,
+		  .len = NL80211_EHT_MAX_CAPABILITY_LEN },
+#endif
 		[NL80211_ATTR_DISABLE_EHT] = { .type = NLA_FLAG },
 		[NL80211_ATTR_MLO_LINKS] =
 		NLA_POLICY_NESTED_ARRAY(nl80211_policy),
